### PR TITLE
#292 Improve Population Diagram styling

### PR DESCRIPTION
**Pull request by Salam-Dalloul**
Before: 
Screenshot 2024-10-03 at 11 52 11 PM

After: 
Screenshot 2024-10-04 at 12 00 40 AM

### CODE REVIEW

**Comment by ddelpiano** [comment]
@stappan could you point us to 2 or 3 populations in production that we can use to validate the implementation done by @Salam-Dalloul ? Thanks

File: frontend/src/components/ProofingTab/GraphDiagram/GraphDiagram.tsx [hunk 1]
```diff
@@ -202,73 +201,116 @@ const processData = (
...
+            
             setTimeout(() => {
                 engine.zoomToFit();
             }, 1000);
             setModelFitted(true);
         }
     }, [modelUpdated, modelFitted, engine]);
-
+    
+    useLayoutEffect(() => {
```
Comment (afonsobspinto) [inline frontend/src/components/ProofingTab/GraphDiagram/GraphDiagram.tsx:293]: Nice. I don't see this everyday 👍

**Comment by stappan** [comment]
@Salam-Dalloul @ddelpiano here are the populations as referenced in GitHub issue #280; that would be impacted by this: https://composer.scicrunch.io/statement/100

File: frontend/src/components/ProofingTab/GraphDiagram/GraphDiagram.tsx [hunk 2]
```diff
@@ -202,73 +201,116 @@ const processData = (
...
+        autoDistribute(engine);
+    }, [engine, modelUpdated, destinations, vias, origins]);
+    
+    useEffect(() => {
+        const currentContainer = containerRef.current;
+        
+        if (modelUpdated && currentContainer) {
+            autoDistribute(engine);
+        }
+    }, [engine, modelUpdated, destinations, vias, origins]);
```
[the remark this answers is not in the document]
Comment (Salam-Dalloul) [reply frontend/src/components/ProofingTab/GraphDiagram/GraphDiagram.tsx:303]: @afonsobspinto actually useLayoutEffect ensures that the diagram is properly shaped on the first render by distributing the layout before the browser renders the page. On the other hand, useEffect handles updates to the diagram when its data changes, ensuring that any modifications are correctly applied after the render. Without useLayoutEffect, the initial layout will not render correctly, and without useEffect, subsequent changes wouldn't be implemented properly.

**Comment by ddelpiano** [comment]
> @Salam-Dalloul @ddelpiano here are the populations as referenced in [GitHub issue #280](https://github.com/MetaCell/sckan-composer/issues/280); that would be impacted by this: https://composer.scicrunch.io/statement/100 https://composer.scicrunch.io/statement/102

@stappan below how kblad 7 and 20 will look like with the smart routing.

**Kblad 7**
screenshot-composer sckan dev metacell us-2024 10 29-11_52_09


**Kblad 20**
screenshot-composer sckan dev metacell us-2024 10 29-11_56_31
screenshot-composer sckan dev metacell us-2024 10 29-11_56_55
screenshot-composer sckan dev metacell us-2024 10 29-11_57_13
screenshot-composer sckan dev metacell us-2024 10 29-11_57_32
screenshot-composer sckan dev metacell us-2024 10 29-11_57_49
screenshot-composer sckan dev metacell us-2024 10 29-11_58_10
screenshot-composer sckan dev metacell us-2024 10 29-11_58_28
screenshot-composer sckan dev metacell us-2024 10 29-11_58_42

**Comment by stappan** [comment]
@ddelpiano the distance between the elements is too big, can we reduce by 40-50%?
does smart routing permit rotation/building the populations laterally (soma or afferent terminal furthest left)

I don't love the really long one kblad20.... it's very difficult to parse; but i don't think shifting it to go left to right will make it easier to view.

**Review by afonsobspinto — APPROVED** [review]

**Pull request closed without merging**